source/adapter.screenshare.js

```diff
@@ -0,0 +1,161 @@
+(function () {
+
+  'use strict';
+
+  var tempGetUserMedia = null;
+
+  if (window.navigator.mozGetUserMedia) {
+    tempGetUserMedia = window.navigator.getUserMedia;
+
+    window.navigator.getUserMedia = function (constraints, successCb, failureCb) {
+      constraints = constraints || {};
+
+      if (constraints.video ? !!constraints.video.mediaSource : false) {
```

Comment:
Following my previous comment (can't remember where), I believe FF and Chrome both accepts 'screen'.
See https://www.webrtc-experiment.com/screen-sharing/
Never tested myself on FF...

I makes more sense to me to use the 'screen' keyword semantically and because it is shared by both browsers.

In any case, It does not change that of the user call mediaSource == 'camera', you should not start the screensharing.